client/app/(tabs)/map.tsx

```diff
@@ -1,19 +1,158 @@
-import { SafeAreaView, StatusBar, StyleSheet, Text } from 'react-native'
-import React from 'react'
+import { StatusBar, StyleSheet, Text, View, Image, TouchableOpacity, Alert } from 'react-native'
+import React, { useEffect } from 'react'
+import MapView, { Callout } from 'react-native-maps'
+import {Marker} from 'react-native-maps';
+import { Ionicons } from '@expo/vector-icons';
+import * as Location from 'expo-location';
+import ChatBubble from '../../components/ChatBubble';
+import { mapThemes } from '../../constants/mapThemes';
+import { useNavigation } from 'expo-router';
+import { NavigationProp } from '@react-navigation/native';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+import useChatBubbles from '../../store/useChatBubbles';
+import useLocation from '../../store/useLocation';
+
+export interface markerLocation {
+  latitude: number,
+  longitude: number,
+}
+
+export interface chatBubble {
+  id: string,
+  chatName: string,
+  chatDesciption: string,
+  chatMembers: number,
+  chatLocation: markerLocation
+}
+
 
 const map = () => {
+  // Map Theme
+  const [ isLightTheme, setIsLightTheme ] = React.useState(true)
+
+  // Map User Location
+  const [location, setLocation] = React.useState({
+    latitude: 37.42182298327982,
+    longitude: -122.0831123739481,
+    latitudeDelta: 0.0043,
+    longitudeDelta: 0.0034
+  })
+
+  // React Native Navigator
+  const navigation = useNavigation<NavigationProp<any>>()
+
+  // Zustand states
+  const { chatBubbles } = useChatBubbles()
+  const { markerLocation } = useLocation()
+  const { setMarkerLocation } = useLocation()
+
+  // User Permissions
+  const [ locationPermission, setLocationPermission ] = React.useState(false)
+
+  const userLocation = async () => {
+    const {status} = await Location.requestForegroundPermissionsAsync()
+    if(status !== "granted"){
+      console.log("Permission to access location was denied")
```

Comment:
Nope, the client don't see the error it self, but because he didn't granted the app with location permissions when he will try to add a new chat he will get an alert that he need to grant location access in the phone settings.